tools/validate_paths.go

```diff
@@ -195,6 +196,21 @@ func checkFiles(knownOC map[string]pathType, files []string) ([]file, error) {
 			errs = append(errs, err.Error())
 		}
 
+		targetFeatureProfileName := getFeatureProfileNameFromPath(f, tmp)
+
+		featureProfileIDName := tmp.GetId().GetName()
+		dependencyList[featureProfileIDName] = make(map[int][]string)
+
+		if targetFeatureProfileName != featureProfileIDName {
+			errs = append(errs, featureProfileIDName+" is inconsistent with path")
+		}
+
+		for _, dependency := range tmp.FeatureProfileDependency {
+			if dependency != nil || dependency.GetName() == "" {
+				dependencyList[featureProfileIDName][len(report)] = append(dependencyList[featureProfileIDName][len(report)], dependency.GetName())
```

Comment:
can you just add a `dependencies []string` field to the `file` type and maintain a separate `validProfile map[string]bool` to keep track of the `featureProfileIDName`s?